src/lib/nested_property.js

```diff
@@ -118,19 +120,38 @@ function npGet(cont, parts) {
  * The ONLY case we are looking for is where the entire array is selected, parts[end] === 'x'
  * AND the replacement value is an array.
  */
-function isDataArray(val, key) {
+// function isNotAContainer(key) {
+//     var containers = ['annotations', 'shapes', 'range', 'domain', 'buttons'];
 
-    var containers = ['annotations', 'shapes', 'range', 'domain', 'buttons'],
-        isNotAContainer = containers.indexOf(key) === -1;
+//     return containers.indexOf(key) === -1;
+// }
 
-    return isArray(val) && isNotAContainer;
+/*
+ * Can this value be deleted? We can delete any empty object (null, undefined, [], {})
+ * EXCEPT empty data arrays. If it's not a data array, it's a container array,
+ * ie containing objects like annotations, buttons, etc
+ */
+var DOMAIN_RANGE = /(^|.)(domain|range)$/;
+function isDeletable(val, propStr) {
```

Comment:
The only use case I'm aware of is animating list of frames `[]` which (serendipitously) functions as a pause button. I've had a bit of trouble with this behavior, so we documented pretty clearly in the web docs that you must use `[null]` instead of `[]` in order to get it passed through.